src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -49,6 +60,57 @@ internal bool AutomaticRedirection
                 _automaticRedirection = value;
             }
         }
+        internal bool UseProxy
+        {
+            get
+            {
+                if (_proxyPolicy == CurlProxyUsePolicy.UseDefaultProxy)
+                {
+                    return false;
+                }
+                return _proxyPolicy != CurlProxyUsePolicy.DoNotUseProxy;
```

Comment:
You should read the code in HttpClientHandler.Windows.cs to fully understand it.

'UseProxy' property is orthogonal to the 'Proxy' property.  But you need to validate invalid combinations of these properties and do so in the .SendAsync() method.

only b) is true.   The problem is your mapping of the Boolean to a 3-state.   You will need to do post-validation and setting in the .SendAsync method in the same way as HttpClientHandler.